packages/datetime/src/components/shortcuts/shortcuts.tsx

```diff
@@ -75,52 +75,69 @@ export interface DatePickerShortcutMenuProps {
-                  )
-                : this.props.shortcuts;
-
-        const shortcutElements = shortcuts.map((shortcut, index) => (
-            <MenuItem
-                active={this.props.selectedShortcutIndex === index}
-                disabled={!this.isShortcutInRange(shortcut.dateRange)}
-                key={index}
-                onClick={this.getShorcutClickHandler(shortcut, index)}
-                shouldDismissPopover={false}
-                text={shortcut.label}
-            />
-        ));
-
-        return (
-            <Menu aria-label="Date picker shortcuts" className={Classes.DATERANGEPICKER_SHORTCUTS} tabIndex={0}>
-                {shortcutElements}
-            </Menu>
-        );
-    }
-
-    private getShorcutClickHandler = (shortcut: DateRangeShortcut, index: number) => () => {
-        const { onShortcutClick } = this.props;
-
-        onShortcutClick(shortcut, index);
-    };
-
-    private isShortcutInRange = (shortcutDateRange: DateRange) => {
-        const { minDate, maxDate } = this.props;
-
-        return isDayRangeInRange(shortcutDateRange, [minDate, maxDate]);
-    };
+export const DatePickerShortcutMenu: React.FC<DatePickerShortcutMenuProps> = props => {
+    const {
+        allowSingleDayRange = false,
+        maxDate,
+        minDate,
+        onShortcutClick,
+        selectedShortcutIndex = -1,
+        timePrecision,
+        useSingleDateShortcuts = false,
+    } = props;
+
+    const shortcuts =
+        props.shortcuts === true
+            ? createDefaultShortcuts(allowSingleDayRange, timePrecision != null, useSingleDateShortcuts)
+            : props.shortcuts;
+
+    return (
+        <Menu aria-label="Date picker shortcuts" className={Classes.DATERANGEPICKER_SHORTCUTS} tabIndex={0}>
+            {shortcuts.map((shortcut, index) => (
+                <ShortcutMenuItem
+                    key={index}
+                    active={selectedShortcutIndex === index}
+                    index={index}
+                    maxDate={maxDate}
+                    minDate={minDate}
+                    onShortcutClick={onShortcutClick}
+                    shortcut={shortcut}
+                />
+            ))}
+        </Menu>
+    );
+};
+
+interface ShortcutMenuItemProps {
+    active: boolean;
+    index: number;
+    maxDate: DatePickerShortcutMenuProps["maxDate"];
+    minDate: DatePickerShortcutMenuProps["minDate"];
+    onShortcutClick: DatePickerShortcutMenuProps["onShortcutClick"];
+    shortcut: DateRangeShortcut;
 }
 
+const ShortcutMenuItem: React.FC<ShortcutMenuItemProps> = props => {
+    const { active, index, maxDate, minDate, onShortcutClick, shortcut } = props;
+
+    const handleClick = React.useCallback(() => onShortcutClick(shortcut, index), [index, onShortcutClick, shortcut]);
+
+    const isShortcutInRange = React.useCallback(
+        (shortcutDateRange: DateRange) => isDayRangeInRange(shortcutDateRange, [minDate, maxDate]),
+        [minDate, maxDate],
+    );
```

Comment:
Now that this is structured this way, I don't think we need a callback here:

```suggestion
    const isShortcutInRange = isDayRangeInRange(shortcut.dateRange, [minDate, maxDate]);
```

Reply:
Ah, good catch! Changed in 9de574d4e882cc11da2e4bbb6ef5afc2410a5aed